components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -151,36 +153,36 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
         private void DiagnosticsOutputTimer_Elapsed(object state) {
             var totalDuration = (DateTime.UtcNow - _diagnosticStart).TotalSeconds;
             _logger.Debug("Identity {deviceId}; {moduleId}", _identity.DeviceId, _identity.ModuleId);
-            _logger.Information("\n   DIAGNOSTICS INFORMATION for Engine : {Name}\n" +
-                "   # Ingestion StartTime - Duration   : {startTime} - {duration} seconds\n" +
-                "   # Ingress DataChanges (from OPC)   : {dataChangesCount} - {dataChangesAverage}/second\n" +
-                "   # Ingress ValueChanges (from OPC)  : {valueChangesCount} - {valueChangesAverage}/second\n" +
-                "   # Ingress BatchBlock buffer size   : {batchDataSetMessageBlockOutputCount}\n" +
-                "   # Encoding Block input/output size : {encodingBlockInputCount}/{encodingBlockOutputCount}\n" +
-                "   # Encoder Notifications processed  : {notificationsProcessedCount}\n" +
-                "   # Encoder Notifications dropped    : {notificationsDroppedCount}\n" +
-                "   # Encoder IoT Messages processed   : {messagesProcessedCount}\n" +
-                "   # Encoder avg Notifications/Message: {notificationsPerMessage}\n" +
-                "   # Encoder avg IoT Message body size: {messageSizeAverage}\n" +
-                "   # Outgress Batch Block buffer size : {batchNetworkMessageBlockOutputCount}\n" +
-                "   # Outgress input buffer count      : {sinkBlockInputCount}\n" +
-                "   # Outgress IoT message count       : {messageSinkSentMessagesCount} - {sentMessagesAverage}/second\n" +
-                "   # Connection retries               : {connectionRetries}\n",
-                Name, _diagnosticStart, _diagnosticStart != DateTime.MinValue ? totalDuration : 0,
-                _messageTrigger.DataChangesCount, _messageTrigger.DataChangesCount / totalDuration,
-                _messageTrigger.ValueChangesCount, _messageTrigger.ValueChangesCount / totalDuration,
-                _batchDataSetMessageBlock.OutputCount,
-                _encodingBlock.InputCount, _encodingBlock.OutputCount,
-                _messageEncoder.NotificationsProcessedCount,
-                _messageEncoder.NotificationsDroppedCount,
-                _messageEncoder.MessagesProcessedCount,
-                _messageEncoder.AvgNotificationsPerMessage,
-                _messageEncoder.AvgMessageSize,
-                _batchNetworkMessageBlock.OutputCount,
-                _sinkBlock.InputCount,
-                _messageSink.SentMessagesCount,
-                _messageSink.SentMessagesCount / totalDuration,
-                _messageTrigger.NumberOfConnectionRetries);
+            if (_messageTrigger.DataChangesCount > 0 || _messageTrigger.ValueChangesCount > 0 || _messageSink.SentMessagesCount > 0)
+            {
```

Comment:
I think this was using structured logging for a reason as Cristian only just recently changed it.   

Also please use .editorconfig formatting as rest of code does.  Do not override formatting settings or else we end up with a hodgepodge of formats in our code.

Reply:
Fixed